service.go

```diff
@@ -201,36 +199,58 @@ func (svc *Service) HandleEvent(ctx context.Context, event *nostr.Event) (result
 			"eventKind": event.Kind,
 			"appId":     app.ID,
 		}).Errorf("Failed to decrypt content: %v", err)
-		return nil, err
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
 	}
 	nip47Request := &Nip47Request{}
 	err = json.Unmarshal([]byte(payload), nip47Request)
 	if err != nil {
-		return nil, err
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+		}).Errorf("Failed to process event: %v", err)
+		return
 	}
 	switch nip47Request.Method {
+	case NIP_47_MULTI_PAY_INVOICE_METHOD:
+		svc.HandleMultiPayInvoiceEvent(ctx, sub, nip47Request, event, app, ss)
+		return
```

Comment:
This is inconsistent with the rest of the methods which return a `resp` and `error` and log those. Would it be more consistent to move the publishing and error logging into the individual Handle... events instead, and make each function return void instead?

Reply:
I think this would be the DRYed up way of doing it as we would have to move all the publish event and error handling to each and every handler (which couldn't be DRYed up further cause it's just 4-5 lines of code anyways)

However I was thinking to change the naming of this to `Handle"AndPublish"MultiPayInvoiceEvent` which might make it look better, wdyt?

Reply:
I think putting "And" in a function name is a bit bad and it more clearly shows the inconsistency between this and the rest of the methods even more. Maybe we can review this with @bumi

Reply:
That is not the only concern, the concern is with the inconsistency in how events are handed which makes the code hard to understand

Reply:
I added a TODO for now. I considered it and think we have more important things to do